### ISSUE TITLE
Dry::Validation::Result#error?(:field) returns false although there is an error (when triggered by rule)

### ISSUE THREAD

**Issue by PragTob** [dry-rb/dry-validation#655]
Hello my friends :wave: : 

Thanks for all your hard work on these gems :green_heart: 

**Describe the bug**

When an error is the result of a rule it seems that `error?(:field)` returns false, although there is an error.

**To Reproduce**

```ruby
require 'dry-validation'

class Nameless < Dry::Validation::Contract
  schema do
    required(:name).filled(:string)
  end

  rule(:name) do
    key.failure('womp womp womp')
  end
end

# change me to empty string/nil to see that error? works in general (for schema)
result = Nameless.new.call(name: 'ABC')

p result # #<Dry::Validation::Result{:name=>"ABC"} errors={:name=>["womp womp womp"]}>

puts "failure? #{result.failure?}" # failure? true
puts "errors: #{result.errors.inspect}" # errors: #<Dry::Validation::MessageSet messages=[#<Dry::Validation::Message text="womp womp womp" path=[:name] meta={}>] options={}>
puts "errors.to_h: #{result.errors.to_h}" # errors.to_h: {:name=>["womp womp womp"]}
puts "result.error?(:name): #{result.error?(:name)}" # result.error?(:name): false
```

**Expected behavior**

`error?` should return `true` because there is an error.

I'm semi worried that because I found `error?` in the `dry-schema` docs that it only works for schema errors but it'd be greatly misleading then. I'd propose to make it work. Failing at that, I'd remove the method from `Validation::Result` as it'd be highly misleading.

**Your environment**

- Affects my production application: kinda (I'm upgrading lots of gems and instead of changing my tests to be `.errors.to_h` and check with `include` using `error?(:field)` seemed like a great way.
- dry-gems versions: all up to date (dry-validation 1.5.2)

<details>

```
bash-5.0# grep 'dry-' Gemfile.lock
    dry-configurable (0.11.6)
      dry-core (~> 0.4, >= 0.4.7)
      dry-equalizer (~> 0.2)
    dry-container (0.7.2)
      dry-configurable (~> 0.1, >= 0.1.3)
    dry-core (0.4.9)
    dry-equalizer (0.3.0)
    dry-inflector (0.2.0)
    dry-initializer (3.0.3)
    dry-logic (1.0.6)
      dry-core (~> 0.2)
      dry-equalizer (~> 0.2)
    dry-schema (1.5.2)
      dry-configurable (~> 0.8, >= 0.8.3)
      dry-core (~> 0.4)
      dry-equalizer (~> 0.2)
      dry-initializer (~> 3.0)
      dry-logic (~> 1.0)
      dry-types (~> 1.4)
    dry-struct (1.3.0)
      dry-core (~> 0.4, >= 0.4.4)
      dry-equalizer (~> 0.3)
      dry-types (~> 1.3)
    dry-types (1.4.0)
      dry-container (~> 0.3)
      dry-core (~> 0.4, >= 0.4.4)
      dry-equalizer (~> 0.3)
      dry-inflector (~> 0.1, >= 0.1.2)
      dry-logic (~> 1.0, >= 1.0.2)
    dry-validation (1.5.2)
      dry-container (~> 0.7, >= 0.7.1)
      dry-core (~> 0.4)
      dry-equalizer (~> 0.2)
      dry-initializer (~> 3.0)
      dry-schema (~> 1.5)
      dry-validation (~> 1)
      dry-types (>= 1.1)
  dry-struct
```
</details>

- Ruby version: 2.6.6
- OS:

```
bash-5.0# uname -a
Linux 441ff575d66a 5.4.0-40-generic #44~18.04.1-Ubuntu SMP Wed Jun 24 23:13:08 UTC 2020 x86_64 Linux
```

(inside an alpine linux container)

--------------------------------------------------------------------------------------------

If you point me the right way/agree this is a bug (and not just a thing of well this _should_ only work for schema) then I'm also happy to work on a fix myself, I know y'all are busy and such :)

Here, have a bunny:

![WhatsApp Image 2020-07-11 at 23 16 53](https://user-images.githubusercontent.com/606517/87770018-bb968780-c81e-11ea-8305-d8405fc87fcb.jpeg)

**Comment by solnic** [comment]
hey @PragTob - I know this is confusing, this is actually designed to only check for schema errors. We should *for sure* change it to take into consideration all errors 😅 This is [why it works the way it works](https://github.com/dry-rb/dry-validation/blob/230138cd190d6ca943173fce730bdd3c042f6fe1/lib/dry/validation/result.rb#L106). I do consider this as a bug though so let's fix it 😄

**Comment by solnic** [comment]
This was addressed via #656
